Scripts/RemediationScripts/Remediate-EnableSSLDBForMySQLFlexibleServer.ps1

```diff
@@ -0,0 +1,801 @@
+
+    $DBForMySQLFlexibleServerDetails | ForEach-Object {
+        if (-not (CheckIfSSLConfigured($_.RequireSecureTransport))) {
+             $DBForMySQLFSWithSSLNotEnabled += $_
+        }
+    }
+   
+     $totalDBForMySQLFSWithSSLNotEnabled = ( $DBForMySQLFSWithSSLNotEnabled  | Measure-Object).Count
+
+    if ( $totalDBForMySQLFSWithSSLNotEnabled -eq 0) {
+        Write-Host "No Azure Database for MySQL flexible server(s) found with SSL disabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$( $totalDBForMySQLFSWithSSLNotEnabled )] Azure Database for MySQL flexible server(s) for which SSL is not enabled." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)	
+
+    $colsProperty = @{Expression = { $_.ResourceName }; Label = "ResourceName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceGroupName }; Label = "ResourceGroupName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceId }; Label = "ResourceId"; Width = 100; Alignment = "left" },
+    @{Expression = { $_.RequireSecureTransport }; Label = "RequireSecureTransport"; Width = 100; Alignment = "left" }
+
+    if (-not $AutoRemediation) {
+        Write-Host "Azure Database for MySQL flexible server(s) with SSL disabled are:"
+         $DBForMySQLFSWithSSLNotEnabled  | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+    }    
+    
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetSSLForDBForMySQLFlexibleServer"
+
+    if (-not (Test-Path -Path $backupFolderPath)) {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "[Step 3 of 4] Back up Azure Database for MySQL flexible server(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($FilePath)) {
+        # Backing up Azure Database for MySQL flexible server(s) details.
+        $backupFile = "$($backupFolderPath)\DBForMySQLFlexibleServerDetailsBackUp.csv"
+         $DBForMySQLFSWithSSLNotEnabled  | Export-CSV -Path $backupFile -NoTypeInformation
+        Write-Host "Azure Database for MySQL flexible server(s) details have been backed up to [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    if (-not $DryRun) {
+        Write-Host $([Constants]::DoubleDashLine)
```

Comment:
No need for double dash line here, as we are not ending the execution at this point.

Double dash line are used when we start and complete the execution of the script. usually before writing remediation summary or in case no resource found for remediation.